src/main/scala/beam/router/r5/CarWeightCalculator.scala

```diff
@@ -0,0 +1,59 @@
+package beam.router.r5
+
+import beam.agentsim.agents.vehicles.BeamVehicleType
+import org.matsim.core.router.util.TravelTime
+
+import java.util.concurrent.ThreadLocalRandom
+import java.util.concurrent.atomic.AtomicInteger
+
+class CarWeightCalculator(workerParams: R5Parameters, travelTimeNoiseFraction: Double = 0D) {
+  private val networkHelper = workerParams.networkHelper
+  private val transportNetwork = workerParams.transportNetwork
+
+  val maxFreeSpeed: Double = workerParams.networkHelper.allLinks.map(_.getFreespeed).max
```

Comment:
Why not just use the helper `networkHelper` already extracted on line 10?

Reply:
Fixed, thanks!